src/magento2/application/skeleton/app/etc/InviqaNonComposerComponentRegistration.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+$pathList[] = realpath(__DIR__ . '/../..') . '/src/*/*/cli_commands.php';
+$pathList[] = realpath(__DIR__ . '/../..') . '/src/*/*/registration.php';
+
+foreach ($pathList as $path) {
+    // Sorting is disabled intentionally for performance improvement
+    $files = glob($path, GLOB_NOSORT);
```

Comment:
This is slow and runs filesystem scanning on each request - can we optimise to have a module manifest?

Reply:
ah ye I remember you did some optimisation for this on toom

Reply:
hmm and I just noticed that we could get rid of this file because it seems now magento has a separate file (registration_globlist.php) which we could put under version control to add the "src" dir to it... probably that would be the default way of doing this nowadays

for the improvement which you did on toom I think that should be something which replaces the default NonComposerComponentRegistration.php provided by magento :think: (maybe they would accept a PR for that in the upstream repo)

Reply:
ah ye I remember you did some optimisation for this one of the projects

hmm and I just noticed that we could get rid of this file because it seems now magento has a separate file (registration_globlist.php) which we could put under version control to add the "src" dir to it... probably that would be the default way of doing this nowadays

for the improvements which you mentioned I think should be something which replaces the default NonComposerComponentRegistration.php provided by magento :think: (maybe they would accept a PR for that in the upstream repo)

Reply:
added the registration_globlist.php now... I tested it and works, but I am not sure if it is overwrited during the inital project setup or not... will do some testing around that. :)

Reply:
well it got overridden probably during composer install :/ will try it with the harness overlay to see if that fixes it

Reply:
well that didn't work either :/

Reply:
ok I will go back to InviqaNonComposerComponentRegistration.php and also add the caching stuff